src/pages/api/appointment/[id].ts

```diff
@@ -0,0 +1,91 @@
+import { getServerAuthSession } from "@server/common/getServerAuthSession";
+import { Appointment } from "@server/db/client";
+import {
+  deleteAppointment,
+  getAppointmentById,
+  updateAppointmentById,
+  updateAppointmentSchema,
+} from "@server/services/appointmentService";
+import { NextApiRequest, NextApiResponse } from "next";
+import { Session } from "next-auth";
+
+const appointmentByIdHandler = async (
+  req: NextApiRequest,
+  res: NextApiResponse
+) => {
+  const { id } = req.query;
+  if (typeof id !== "string") {
+    res.status(400).json({ message: "Invalid Appointment ID." });
+    return;
+  }
+
+  const session = await getServerAuthSession({ req, res });
+  if (!session) {
+    res.status(403).json({ message: "Forbidden." });
+    return;
+  }
+
+  let appointment: Appointment | null;
+
+  switch (req.method) {
+    case "GET":
+      appointment = await getAppointmentById(id);
+      if (appointment) {
+        if (isAuthorized(session, appointment)) {
+          res.status(200).json(appointment);
+        } else {
+          res.status(403).json({ message: "Forbidden." });
+        }
+      } else {
+        res.status(404).json({ message: "Appointment not found." });
+      }
+      break;
+
+    case "PATCH":
+      const patch = updateAppointmentSchema.safeParse(req.body);
```

Comment:
I think you will need JSON.parse() here otherwise we will get a ZodError when we parse the req.body

Reply:
Same for every other api

Reply:
Not needed. You didn't set the headers to Next was not automatically parsing the body.

> req.body - An object containing the body parsed by content-type, or null if no body was sent

https://nextjs.org/docs/api-routes/request-helpers